packages/react-data-grid-react-window/src/components/DataGrid/renderDataGrid.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import {
+  DataGridContextValues,
+  DataGridState,
+  renderDataGrid_unstable as baseRender,
+} from '@fluentui/react-components';
+import { HeaderRefContextProvider } from '../../contexts/headerRefContext';
+import { BodyRefContextProvider } from '../../contexts/bodyRefContext';
+
+/**
+ * Render the final JSX of DataGrid
+ */
+export const renderDataGrid_unstable = (
+  state: DataGridState,
+  contextValues: DataGridContextValues
+) => {
+  const headerRef = React.useRef<HTMLDivElement | null>(null);
+  const bodyRef = React.useRef<HTMLDivElement | null>(null);
```

Comment:
Please use hooks in the state hook (i.e. `useDataGrid` hook) everything inside a render function should be a pure function

Reply:
Ah I didn't realize it should be pure, updated now